### PR TITLE
Add Form Doctor

### PR DESCRIPTION
**Pull request by cvalencia1991**
# In this milestone, I did the following activities 

- Create Add doctor Reducer and action
- Create a form in the component Add Doctor 
- Add Dispatch Method to send it to the API `http://localhost:3000/api/v1/doctors`
- Fix Linters Errors

### CODE REVIEW

**Review by Camilovelag — APPROVED** [review]
## APPROVED!
### Highlights 🥇
- Checked last react-router-dom best practices (setup `router` in the entry point file rather `App.js` file).
### _Optional_
- Check [createSlice](https://redux-toolkit.js.org/api/createSlice) method from _redux toolkit_ to create your reducer.

Happy coding! 👋

**Review by asdt560 — CHANGES_REQUESTED** [review]
Great work on the reducers, but I very strongly suggest using the CreateSlice method and syntax for ease of working with the state. 

As well as sticking to lowercase for function names (because React Components must be uppercase) and folder names to avoid ambiguity.

I have a possible code you can use for this:
doctorsSlice.js:

    import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
    const getDoctorsData = createAsyncThunk(
      'doctors/getDoctorsData',
      async () => {
      const resp = await fetch('http://localhost:3000/api/v1/doctors')
        .then((resp) => resp.json())
        .then((result) => result);
      return resp;
      },
    );

    const addDoctor = createAsyncThunk('doctors/addDoctor', async (obj) => {
      const response = await fetch('http://localhost:3000/api/v1/doctors', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(obj),
      });
      return response.json();
    });

    const deleteDoctor = createAsyncThunk(
      'doctors/deleteDoctor',
      async (id) => {
        const resp = await fetch(`http://localhost:3000/api/v1/doctors/${id}`, { method: 'Delete' })
          .then((resp) => resp.json())
          .then((result) => result);
        return resp;
      },
    );

    const doctorsSlice = createSlice({
      name: 'greeting',
      initialState: {
        loading: false,
        doctors: [],
      },
      extraReducers: (builder) => {
        builder.addCase(getDoctorsData.pending, (state) => ({
           ...state,
          loading: true,
        }));
        builder.addCase(getDoctorsData.fulfilled, (state, action) => ({
          ...state,
          loading: false,
          doctors: action.payload,
        }));
        builder.addCase(getDoctorsData.rejected, (state, action) => ({
          ...state,
          loading: false,
          doctors: [],
          error: action.error.message,
        }));
        builder.addCase(deleteDoctor.pending, (state) => ({
          ...state,
          loading: true,
        }));
       builder.addCase(deleteDoctor.fulfilled, (state, action) => ({
          ...state,
          loading: false,
          doctors: [
             ...state.doctors.filter((doctor) => doctor.id !== action.payload.id),
          ],
       }));
       builder.addCase(deleteDoctor.rejected, (state, action) => ({
          ...state,
         loading: false,
         error: action.error.message,
      }));
      builder.addCase(addDoctor.pending, (state) => ({
           ...state,
           loading: true,
       }));
       builder.addCase(addDoctor.fulfilled, (state) => ({
          ...state,
          loading: false,
       }));
      builder.addCase(addDoctor.rejected, (state, action) => ({
           ...state,
           loading: false,
          error: action.error.message,
      }));
      },
    });

    export default doctorsSlice.reducer;
    export { getDoctorsData, deleteDoctor, addDoctor };

File: src/redux/Doctors/doctor.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(obj),
+  });
+  return response.json();
+});
+
+export const DeleteDoctor = createAsyncThunk(DEC_DOC, async (id) => {
```
Comment (asdt560) [inline src/redux/Doctors/doctor.js:39]: For example, this function being named DeleteDoctor in upper case causes trouble with the DeleteDoctor React component, to which this function needs to be imported.

**Comment by cvalencia1991** [comment]
Thank you :smile:

**Pull request merged**